Packs/CortexXDR/Integrations/XDR_iocs/XDR_iocs.py

```diff
@@ -624,29 +721,40 @@ def module_test(client: Client):
 
 def fetch_indicators(client: Client, auto_sync: bool = False):
     demisto.debug("fetching IOCs: starting")
-    if not get_integration_context() and auto_sync:
-        demisto.debug("fetching IOCs: running sync with first_time=True")
-        # this will happen on the first time we run
-        xdr_iocs_sync_command(client, first_time=True)
+    integration_context = get_integration_context()
+    demisto.debug(f"The integration context inside fetch_indicators is {integration_context=}")
+    if (((not integration_context) or (integration_context.get('is_first_sync_phase', False)))
+            and auto_sync):
+        if not integration_context:
+            sync_time = datetime.now(timezone.utc)
+            update_integration_context(update_sync_time_with_datetime=sync_time, update_is_first_sync_phase='true')
+        demisto.debug("fetching IOCs: running sync with is_first_stage_sync=True")
+        xdr_iocs_sync_command(client=client, is_first_stage_sync=True, called_from_fetch=True)
     else:
-        # This will happen every fetch time interval as defined in the integration configuration
+        # This will happen every fetch time interval as defined in the integration configuration and is_first_sync_phase=False
         demisto.debug("fetching IOCs: running get_changes")
-        get_changes(client)
         if auto_sync:
-            demisto.debug("fetching IOCs: auto_sync is on")
+            demisto.debug("fetching IOCs from xsoar: auto_sync is on")
             tim_insert_jsons(client)
             demisto.debug("checking if iocs_to_keep should run")
             if is_iocs_to_keep_time():
```

Comment:
we need to clean all this iocs_to_keep code

Reply:
should i also deprecate the xdr-iocs-to-keep-file command?